basics/foreach.md

```diff
@@ -0,0 +1,105 @@
+# Foreach
+
+{{#img-right}}dman-teacher-foreach.jpg{{/img-right}}
+
+Mit der `foreach`-Schleife besitzt D ein Konstrukt, das 
+Iterationen besser lesbar und weinger fehlerträchtig macht.
+
+### Elementweise Iteration
+
+Ein gegebenes Array `arr` vom Typ `int[]` kann mithilfe der 
+`foreach`-Schleife elementweise iteriert werden:
+
+    foreach (int e; arr) {
+        writeln(e);
+    }
+
+Das erste Feld der `foreach`-Definition ist der Name der in der 
+Schleifen-Iteration benutzte Variable. Ihr Typ wird automatisch
+abgeleitet:
+
+    foreach (e; arr) {
+        // typeof(e) == int
+        writeln(e);
+    }
+
+Das zweite Feld muss ein Array sein - oder ein spezielles 
+iterierbares Objekt, das **Range** genannt wird. Es wird im 
+nächsten Abschnitt behandelt. 
+
+### Zugriff über Referenz
+
+Elemente des Arrays oder der Range werden während des Iterierens
+kopiert. Für Basistypen ist die akzeptabel, für große Typen mag
+das aber ein Problem darstellen. Um das Kopieren zu verhindern,
+oder um die Elemente *in-place* zu ändern, kann `ref` verwendet 
```

Comment:
Ich würde versuchen möglichst wenig Fremdwörter zu verwenden und wenn auch gleich zu erklären, e.g.
- in-place (in ihren Speicherzellen)
- in ihren Speicherzellen (engl.: *in-place*)